lib/R/Mbio/abundance.R

```diff
@@ -0,0 +1,52 @@
+#   # Inputs
+#   ranking_method <- 'Median'
+#   taxonomicLevel <- 'Class'
+#   verbose <- T
+
+#   computeMessage <- ''
+
+#   # This part seems redundant, but it's to get around using tidyr::complete (see shiny's otu_table.R line 18)
+#   otu <- makeOTU(df, taxonomicLevel)
+#   plot.data::logWithTime(paste("Created OTU table with", NROW(otu), "samples and", (NCOL(otu)-1), "taxa."), verbose)
+
+#   # Reshape back to sample, taxonomicLevel, abundance
+#   formattedDT <- data.table::melt(otu, measure.vars=colnames(otu)[-1], variable.factor=F, variable.name=taxonomicLevel, value.name="Abundance")
+
+#   # Rank by method
+#   rankingMethods <- c('median','max','q3','var')
+#   rankedTaxaList <- lapply(rankingMethods, topNTaxa, df=formattedDT, cutoff=10, taxonomicLevel = taxonomicLevel)
+
+#   # Admittedly terrible. Functions but needs clenaing up. Struggled to find a nice data.table solution
```

Comment:
so assuming appending the ranking method to the column header is the method we ultimately use (which seems to be what youre doing here).. can topNTaxa return a data.table w those columns and the ranking method used already appended, rather than just a list of taxa?

Reply:
it might even take the otu table format, rather than the reshaped thing, do whatever shaping is necessary to answer the question for itself, and then return the otu table format as well?

Reply:
.. also possible ive misunderstood this code, not highly readable yet

Reply:
re column names: i fought this for a while but came up with the above because I wanted to know the original column names as well since those would be used as the display names. I could also remove the appended info after appending it in the topNTaxa but that seemed not great either. 

Additionally i'm not sure appending the method is the best way to go about it anyhow. I also wrangled with the idea of returning the `rankedTaxaList` as computed variables, but that would require a more complicate api. However, it would be really nice to just return a list with one element per method that contains the top N taxa. Then, the output dt would only need to be the unique taxa that are in at least one of the top N lists. For example, in the bee data, it was something like only 15 unique taxa showed up in any of the four top N lists.

Reply:
thats clever. i think id be fine w it. were already anticipating three endpoints for each compute, for the reason that these can be very variable from one compute to the next and also quite complicated. so im not opposed to a complicated api if that is also the best api. the dt i suppose would be returned with our data endpiont or whatever it ends up called, then the ranked taxa list list could be part of the metadata endpoint. and the app in the data service just needs to know about both of them, and how they relate.